### PR TITLE
Added can_blaze property to /sites/{siteId} endpoint

### PR DESCRIPTION
**Pull request by paulopmt1**
Fixes 1532-gh-Automattic/dotcom-forge

## Proposed changes:

Added `can_blaze` property to /sites/{siteId} endpoint (inside options attribute), so Calypso can understand when to allow users to use it. We're also checking it by each `blog_id`, so we don't need to rely on the primary blog info.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

More context to this change can be found here: p1679438980595779/1679438966.648259-slack-C04DZ725FEK

## Testing instructions:

* Apply this PR change in your sandbox with the command: `bin/jetpack-downloader test jetpack add/new-has-promote-widget-property-to-endpoint`
* Make sure your example site has En locale and it's published
* Open [your WordPress console](https://developer.wordpress.com/docs/api/console/), access the `/sites/{siteId}?http_envelope=1` wpcom v1.1 endpoint, and test if the `options->can_blaze` attribute is true
* Change your example site to a PT-br locale
* The `options->can_blaze` attribute should now be false
* Test with Simple, Atomic and Self-hosted sites, by installing jetpack beta plugin and follow these instructions: pb5gDS-2kZ-p2#comment-3156 and applying the code of `add/new-has-promote-widget-property-to-endpoint` branch on them

image

You can also test it using [the complement Calypso PR](https://github.com/Automattic/wp-calypso/pull/74795)

## Does this pull request change what data or activity we track or use?

No.

### CODE REVIEW

**Review by lupus2k — DISMISSED** [review]
LGTM!

**Review by lupus2k — DISMISSED** [review]
LGTM after rebase ;)

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php
Comment (dlind1) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: I'm wondering if it shouldn't be `jetpack_dsp_promote_posts_enabled`?

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php
Comment (dlind1) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Would the `has_promote_widget` be removed from the `/me/` endpoint in a separate PR?

**Review by jeherve — CHANGES_REQUESTED** [review]
This tests well for me on WordPress.com simple, where we have the function hooking into `dsp_promote_posts_enabled`, but that will not work on WoA or self-hosted sites I'm afraid, since we do not have anything hooking into the filter there.

I think we could do away with the filter at this point, since we've since developed other means of querying for blaze status. I believe something like this should work in all environments for example:

```diff
diff --git a/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php b/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php
index 98c5f9b1ac..52f28fb24f 100644
--- a/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php
+++ b/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php
@@ -874,7 +874,7 @@ class WPCOM_JSON_API_GET_Site_Endpoint extends WPCOM_JSON_API_Endpoint {
 					$options[ $key ] = $site->get_wpcom_staging_blog_ids();
 					break;
 				case 'has_promote_widget':
-					$options[ $key ] = apply_filters( 'dsp_promote_posts_enabled', false, $site->blog_id );
+					$options[ $key ] = $site->can_blaze();
 					break;
 			}
 		}
diff --git a/projects/plugins/jetpack/sal/class.json-api-site-base.php b/projects/plugins/jetpack/sal/class.json-api-site-base.php
index 0da4dcc025..e016998a06 100644
--- a/projects/plugins/jetpack/sal/class.json-api-site-base.php
+++ b/projects/plugins/jetpack/sal/class.json-api-site-base.php
@@ -11,6 +11,7 @@
  * @package automattic/jetpack
  **/
 
+use Automattic\Jetpack\Blaze;
 use Automattic\Jetpack\Status;
 use Automattic\Jetpack\Status\Host;
 
@@ -1500,4 +1501,13 @@ abstract class SAL_Site {
 	public function get_wpcom_staging_blog_ids() {
 		return get_option( 'wpcom_staging_blog_ids', array() );
 	}
+
+	/**
+	 * Get the site's Blaze eligibility status.
+	 *
+	 * @return bool
+	 */
+	public function can_blaze() {
+		return (bool) Blaze::site_supports_blaze( $this->blog_id );
+	}
 }
```

Maybe we could also use this opportunity to move away from `has_promote_widget` as well, since  we moved away from the `promote` nomenclature to `blaze`? That would require some changes in Calypso, but we're making changes anyway. Maybe the new key could be `can_blaze`, for example?

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php
Comment (paulopmt1) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: That's a good point, moved away [from promote_widget name here](https://github.com/Automattic/jetpack/pull/29614/commits/33a80ca0bbe903d524e89ebc9177b881acb77bfd), using `can_blaze` instead.

**Review by jeherve — APPROVED** [review]
This tests well for me on WordPress.com and on self-hosted. 👍 

Feel free to get a review from someone on your team though, I'm a bit biased with that code :)

**Review by lupus2k — APPROVED** [review]
LGTM 🎉

**Comment by paulopmt1** [comment]
Thanks for your reviews and suggestions, folks!

**Pull request merged**
